server/socket/index.ts

```diff
@@ -1,147 +1,104 @@
-          ws.expirationTimer = global.setTimeout(() => {
-            // TODO: do i need to manually close the socket here?
-            log.trace(`Websocket ${sessionId} expired, emitting close`);
-            wss.emit(WebsocketEventType.Close);
-          }, EXPIRATION_TIME);
-
-          // get user if verified
-          if (payload.jwt) {
-            jwt.verify(payload.jwt, JWT_SECRET, async (err, decodedUser) => {
-              if (err) {
-                log.info(`received message with malformed JWT: ${payload.jwt}`);
-              } else {
-                ws.isAuthenticated = true;
-                ws.user = decodedUser;
-              }
-            });
-          }
-        } else {
-          log.error('received malformed message');
-        }
-      } catch (e) {
-        log.error('received malformed message');
-      }
-    });
-
-    ws.on(WebsocketEventType.Close, () => {
-      log.trace(`Received close event for websocket ${sessionId}`);
-      if (sessionMap[sessionId]) {
-        delete sessionMap[sessionId];
-      }
-      if (userId && userMap[userId]) {
-        delete userMap[userId];
-      }
-    });
-  });
-
-  // TODO: maybe unify these, or else remove the event type from payload and add it manually here?
-  wss.on(WebsocketMessageType.InitializeScrollback, (payload: IWebsocketsPayload<any>, userIds: number[]) => {
-    if (logging) log.info(`Payloading ${JSON.stringify(payload)} to users ${JSON.stringify(userIds)}`);
-    for (const user of userIds) {
-      if (user && user in userMap && userMap[user].isAlive) {
-        userMap[user].send(JSON.stringify(payload));
-      }
-    }
-  });
-
-  wss.on(
-    WebsocketMessageType.Notification,
-    (payload: IWebsocketsPayload<any>, notifications: { [user: number]: any }) => {
-      if (logging) {
-        log.info(`Payloading ${JSON.stringify(payload)} to users ${JSON.stringify(Object.keys(notifications))}`);
-      }
-      for (const [ user, notification ] of Object.entries(notifications)) {
-        if (user && user in userMap && userMap[user].isAlive) {
-          // augment notification with unique database id
-          payload.data.id = notification.id;
-          payload.data.subscription_id = notification.subscription_id;
-          userMap[user].send(JSON.stringify(payload));
-        }
-      }
-    }
+export function setupWebSocketServer(httpServer: http.Server) {
+	// since the websocket servers are not linked with the main Commonwealth server we do not send the socket.io client
+// library to the user since we already import it + disable http long-polling to avoid sticky session issues
+	const io = new Server(httpServer, { transports: ['websocket'], cors: {
+			origin,
+			methods: ["GET", "POST"]
+		}});
+
+	io.use(authenticate);
+
+	io.on('connection', (socket) => {
+		log.trace(`${socket.id} connected`)
+		socket.on('disconnect', () => {
+			log.trace(`${socket.id} disconnected`);
+		});
+	});
+
+	io.engine.on('connection_error', (err) => {
```

Comment:
Is handling reconnects something that happens automatically? Or how do we deal with that?

Reply:
Reconnects are handled automatically by Socket.io